tools/custodian-container/main.go

```diff
@@ -0,0 +1,242 @@
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// Create the bind mounts for input/output
+func GenerateBinds(args []string, outputPath string, policyPath string) []string {
+	// Policy
+	policy, err := filepath.Abs(policyPath)
+	if err != nil {
+		log.Fatalf("Unable to load policy. %v", err)
+	}
+
+	containerPolicy := CONTAINER_HOME + filepath.Base(policy)
+	binds := []string{
+		policy + ":" + containerPolicy + ":ro",
+	}
+
+	// Output Path
+	if outputPath != "" {
+		outputPath, err = filepath.Abs(outputPath)
+		if err != nil {
+			log.Fatalf("Unable to parse output path. %v", err)
+		}
+
+		binds = append(binds, outputPath + ":" + CONTAINER_HOME + "output:rw")
+	}
+
+	// Azure CLI support
+	azureCliConfig := GetAzureCliConfigPath()
+	if azureCliConfig != "" {
+		// Bind as RW for token refreshes
+		binds = append(binds, azureCliConfig + ":" + CONTAINER_HOME + ".azure:rw")
+	}
+
+	// AWS config
+	awsConfig := GetAwsConfigPath()
+	if awsConfig != "" {
+		binds = append(binds, awsConfig + ":" + CONTAINER_HOME + ".aws:ro")
+	}
+
+	return binds
+}
+
+// Fix the policy arguments
+func SubstitutePolicy(args []string) string {
+	originalPolicy := args[len(args)-1]
+	args[len(args)-1] = CONTAINER_HOME+filepath.Base(originalPolicy)
+
+	return originalPolicy
+}
+
+// Fix the output arguments
+func SubstituteOutput(args []string) string {
+	var outputPath string
+
+	for i := range args{
+		arg := args[i]
+		if arg == "-s" || arg == "--output-dir" {
+			outputPath = args[i+1]
+
+			if !(strings.HasPrefix(outputPath, "s3://") ||
+				strings.HasPrefix(outputPath, "azure://") ||
+				strings.HasPrefix(outputPath, "gs://")) {
+				args[i+1] = CONTAINER_HOME + "output"
+				return outputPath
+			}
+		}
+	}
+
+	return ""
+}
+
+// Get list of environment variables
+func GenerateEnvs() []string {
+	var envs []string
+
+	// Bulk include matching variables
+	var re = regexp.MustCompile(`^AWS|^AZURE_|^MSI_|^GOOGLE`)
+	for _, s := range os.Environ() {
+		if re.MatchString(s) {
+			envs = append(envs, s)
+		}
+	}
+
+	return envs
+}
+
+// Find Azure CLI Config if available so
+// we can mount it on the container.
+func GetAzureCliConfigPath() string {
+	// Check for override location
+	azureCliConfig := os.Getenv("AZURE_CONFIG_DIR")
+	if azureCliConfig != "" {
+		return filepath.Join(azureCliConfig, "config")
+	}
+
+	// Check for default location
+	var configPath string
+
+	if runtime.GOOS == "windows" {
+		configPath = filepath.Join(os.Getenv("USERPROFILE"), ".azure")
+	} else {
+		configPath = filepath.Join(os.Getenv("HOME"), ".azure")
+	}
+
+	if _, err := os.Stat(configPath); err == nil {
+		return configPath
+	}
+
+	return ""
+}
+
+// Find AWS Config if available so
+// we can mount it on the container.
+func GetAwsConfigPath() string {
+	var configPath string
+
+	if runtime.GOOS == "windows" {
+		configPath = filepath.Join(os.Getenv("USERPROFILE"), ".aws")
+	} else {
+		configPath = filepath.Join(os.Getenv("HOME"), ".aws")
+	}
```

Comment:
these are both good for happy path on the majority, there are technically environment variables which let the contents of this dir be specifcied in different locations,  per the docs for now i'm fine omitting, but wanted to call out for awareness. the change here would be updating the interface to returning a mapping of source: target mounts as there are potentially multiple. i vaguely remember issues with mounting single files in the past though.

```
AWS_SHARED_CREDENTIALS_FILE – Specifies the location of the file that the AWS CLI uses to store access keys (the default is ~/.aws/credentials).
AWS_CONFIG_FILE – Specifies the location of the file that the AWS CLI uses to store configuration profiles (the default is ~/.aws/config).
```

Reply:
Oh ok great #4178